### ISSUE TITLE
Replace with empty string

### ISSUE THREAD

**Issue by schreddies** [semgrep/semgrep#5674]
**Describe the bug**
Hi, 
During experiments with experimental feature - autofix, I got problem with fixing something with removing the string, or replacing it with empty string.

**To Reproduce**
``` 
fix-regex:
        regex: 'fmt(.*)\('
        replacement: ''
```
Error message from Semgrep:
`'regex' and 'replacement' values required when using 'fix-regex'`
Using following config:
```
fix-regex:
        regex: 'fmt(.*)\('
        replacement: '\'
```
Response from Semgrep:
`unable to use regex to modify file asdf.go with fix 'None': bad escape (end of pattern) at position 0`

**Expected behavior**
The replacement with value `''` should be possible, with empty value.
For now I can use the Zero Width Space `U+200B`, but it might be non-compatible for many use-cases. 

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [X] P2: regular bug that should get fixed

**Environment**
Newest Semgrep version

**Use case**
Trying to use autofix feature to fix security issues by removing them :)

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1596/replace-with-empty-string. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by IagoAbal** [comment]
cc @mmcqd

**Comment by schreddies** [comment]
@IagoAbal @mmcqd the 'issue' is still there, but I am closing this one, as it is working for me (Zero width space)

**Comment by IagoAbal** [comment]
Maybe this would work now with @nmote's latest AST-based autofix?

**Comment by nmote** [comment]
I don't think AST-based autofix would affect it. It doesn't affect fix-regex at all, since fix-regex is by definition text-based. I bet this is due to a Python truthiness test somewhere like `if replacement:`, since the empty string is falsy. I bet that finding that spot and replacing `if replacement:` with `if replacement is not None:` would address this issue.

**Comment by nmote** [comment]
Fixed by #7772
